### PR TITLE
Prevent bogus mention notifications (fixes #5300)

### LINKED ISSUE #5300: mention bug
member names problem with the same name

sample;

member A : kelvin

member B : kelvin klen

see other person (@ kelvin klen)  mention notifications on their panel  "@ kelvin"

-----------------------------------------------------------------------

@Antes  hocam aynı isimle başlayan kullanıcı adları bahsetme bildirimlerinde sorun yaratıyor 

örnegin "mehmet" isimli kullanıcı  bir konuda "mehmet ali" isimli kullanıcıdan bahsedildiğinde kendi uyarı paneline sizden bahsedildi uyarısını görüyor  halbuki bahsedilen   "@ mehmet ali"

### Steps to reproduce
1. 
2. 

### Environment (complete as necessary)
- Version/Git revision:
- Database Type:
- Database Version:
- PHP Version:

### Additional information/references

### PR DESCRIPTION
**Pull request by Oldiesmann**
Also added a few comments to explain the code

Signed-off-by: Michael Eshom <oldiesmann@oldiesmann.us>

### CODE REVIEW

**Comment by live627** [comment]
Check your white space, please.

File: Sources/Subs-Post.php [hunk 1]
```diff
@@ -1728,9 +1728,20 @@ function createPost(&$msgOptions, &$topicOptions, &$posterOptions)
 
 	if (!empty($modSettings['enable_mentions']))
 	{
+	    // Get any members who were possibly mentioned
```
Comment (live627) [inline Sources/Subs-Post.php]: Still spaces here

**Comment by Sesquipedalian** [comment]
I don't think we really need to use preg_match() for this. Since the string is fixed, a strpos() check should do the job much more efficiently.

File: Sources/Subs-Post.php [hunk 2]
```diff
@@ -1738,7 +1738,7 @@ function createPost(&$msgOptions, &$topicOptions, &$posterOptions)
 			// Remove any members who weren't actually mentioned, to prevent bogus notifications
 			foreach ($msgOptions['mentioned_members'] as $m)
 			{
-				if (preg_match('~\[member=' . $m['id'] . '\]@' . $m['real_name'] . '\[/member\]~u', $msgOptions['body']) === 0)
+				if (strpos('[member=' . $m['id'] . ']' . $m['real_name'] . '[/member]', $msgOptions['body']) === false)
```
Comment (Sesquipedalian) [inline Sources/Subs-Post.php]: It looks like you accidentally removed the @ symbol from the string here.

**Pull request merged**